src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestProducer.kt

```diff
@@ -0,0 +1,59 @@
+package org.rust.cargo.runconfig.test;
+
+import com.intellij.execution.actions.ConfigurationContext
+import com.intellij.execution.actions.RunConfigurationProducer
+import com.intellij.openapi.util.Ref
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.impl.RustFnItemImpl
+
+class CargoTestProducer : RunConfigurationProducer<CargoTest>(CargoTestType()) {
+    override fun isConfigurationFromContext(configuration: CargoTest?, context: ConfigurationContext?): Boolean {
+        return configuration?.name == configurationName(context)
+    }
+
+    override fun setupConfigurationFromContext(configuration: CargoTest?,
```

Comment:
I think you can omit `?` for all arguments here. The base function does not have nullability annotations, so Kotlin conservatively uses `?` here. However the docstring does not mentions `null` at all, so I think that all these are in fact `NotNull`. 

This in an important gotcha when overriding a Java method in Kotlin. If you see that all parameters are nullable it usually means missing annotations. You need to figure out right nullability manually in such case.